docs/src/literate-tutorials/hyperelasticity.jl

```diff
@@ -417,14 +420,53 @@ function solve()
     return u
 end
 
+# Alternatively, it is possible to solve the same system with NonlinearSolve by defining a few wrappers:
+# Note that as written, this is missing some of the optimizations that you would want for a real deployment.
+function nonlinearsolve_solve()
+    dh, cv, fv, mp, ΓN, dbcs = create_grid()
 
-# Run the simulation
+    ## Create sparse matrix and residual vector
+    K = create_sparsity_pattern(dh)
+    g = zeros(ndofs(dh))
+
+    function nl_assemble(g, u, p)
+        (;K, dh, cv, fv, mp, ΓN, dbcs) = p
+        apply!(u, dbcs)
+        assemble_global!(K, g, dh, cv, fv, mp, u, ΓN)
+        apply_zero!(K, g, dbcs)
+        g
+    end
+    function nl_assemble_jac(K, u, p)
+        (;g, dh, cv, fv, mp, ΓN, dbcs) = p
```

Comment:
What's the reason for passing `ΓN` etc through `p` instead of just directly using them from the scope above (since this is a closure closing over it)?